osquery/tables/system/windows/shortcut_files.cpp

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/utils/conversions/join.h>
+
+#include <osquery/logger/logger.h>
+#include <osquery/utils/windows/shelllnk.h>
+
+#include <boost/filesystem.hpp>
+
+#include <sstream>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+QueryData genShortcutFiles(QueryContext& context) {
+  QueryData results;
+  auto paths = context.constraints["path"].getAll(EQUALS);
+  // Expand contraints
+  context.expandConstraints(
+      "path",
+      LIKE,
+      paths,
+      ([&](const std::string& pattern, std::set<std::string>& out) {
+        std::vector<std::string> patterns;
+        auto status =
+            resolveFilePattern(pattern, patterns, GLOB_ALL | GLOB_NO_CANON);
+        if (status.ok()) {
+          for (const auto& resolved : patterns) {
+            out.insert(resolved);
+          }
+        }
+        return status;
+      }));
+
+  boost::system::error_code ec;
+  for (const auto& lnk : paths) {
```

Comment:
The logic in this for loop is large. I usually recommend breaking loops over 100 lines long into smaller chunks to help maintain what should/should not remain in scope.

Reply:
So this loop is basically responsible for parsing the lnk file.  
I split the actual lnk data parsing into another function `parseShortcutFiles`  
The loop now just reads the lnk file and verify its shortcut file, if it is it calls `parseShortcutFiles` to parse the data  
Is this what you r looking for?